src/org/ohdsi/usagi/ReadCodeMappingsFromFile.java

```diff
@@ -77,14 +77,18 @@ private void readNext() {
 							buffer.setMappingStatus(MappingStatus.INVALID_TARGET);
 							buffer.setComment("Invalid existing target: " + row.get("conceptId"));
 						} else {
-							// Type and provenance might not be available in older Usagi files
+							String mappingTypeString = row.get("mappingType", "MAPS_TO");
+							if (!mappingTypeString.startsWith("MAPS_TO")) {
+								// For backwards compatibility with files using event/value/unit
+								mappingTypeString = mappingTypeString
+										.replace("EVENT", "MAPS_TO")
+										.replace("VALUE", "MAPS_TO_VALUE")
+										.replace("UNIT", "MAPS_TO_UNIT");
+							}
```

Comment:
This could be replaced by a switch method in MappingTarget.Type
```java
enum Type {
...
public static Type valueOfCompat(String value) {
  switch (value) {
    case "EVENT": return MAPS_TO;
    case "VALUE": return MAPS_TO_VALUE;
    case "UNIT": return MAPS_TO_UNIT;
    default: return valueOf(mappingTypeString);
  }
}

}
```